### PR TITLE
Do static configuration of ironic.conf when building the image

### PR DESCRIPTION
**Pull request by dtantsur**
This is consistent with ironic-inspector and results in a more complete
image. The options are sorted for readability.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,42 @@ RUN yum install -y python-requests && \
 RUN mkdir /tftpboot && \
     cp /usr/share/ipxe/undionly.kpxe /usr/share/ipxe/ipxe.efi /tftpboot/
 
+RUN cp /etc/ironic/ironic.conf /etc/ironic/ironic.conf_orig && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT auth_strategy noauth && \
```
Comment (markmc) [inline Dockerfile]: Appreciate you're just copying existing code around, but wouldn't using `crudini --merge` be more readable?

```
crudini --merge /etc/ironic/ironic.conf <<EOF
[DEFAULT]
auth_strategy=noauth
debug=true

[agent]
deploy_logs_collect=always
EOF
```

Comment (dtantsur) [reply Dockerfile]: OH, I've just been wondering if something like that exists, thanks Mark!

**Comment by metal3ci** [comment]
Build ABORTED, see build http://10.8.144.11:8080/job/dev-tools/847/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/851/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/854/

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/861/

**Comment by juliakreger** [comment]
Disregard last comment, I finally read my email.

**Pull request merged**